packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasSummaryStore.ts

```diff
@@ -0,0 +1,238 @@
+    token, // when the token changes
+    l1Provider,
+    l2Provider,
+    walletAddress, // when user switches account or if user is not connected
+    setGasSummaryStatus
+  ])
+
+  // Estimated L1 gas fees, denominated in Ether, represented as a floating point number
+  const estimatedL1GasFees = useMemo(() => {
+    const gasPrice = isDepositMode ? l1GasPrice : l2GasPrice
+    return parseFloat(utils.formatEther(result.estimatedL1Gas.mul(gasPrice)))
+  }, [isDepositMode, l1GasPrice, l2GasPrice, result.estimatedL1Gas])
+
+  // Estimated L2 gas fees, denominated in Ether, represented as a floating point number
+  const estimatedL2GasFees = useMemo(
+    () =>
+      parseFloat(
+        utils.formatEther(
+          result.estimatedL2Gas
+            .mul(l2GasPrice)
+            .add(result.estimatedL2SubmissionCost)
+        )
+      ),
+    [result.estimatedL2Gas, l2GasPrice, result.estimatedL2SubmissionCost]
+  )
+
+  useEffect(() => {
+    // Since we are using a debounced value, it's possible for the value to be outdated
+    // Wait for it to sync before running the gas estimation
+    if (!amountDebounced.eq(amount)) {
+      setGasSummaryStatus('loading')
+      return
+    }
+
+    if (!isDepositMode) {
+      if (
+        isTokenArbitrumOneNativeUSDC(token?.address) ||
+        isTokenArbitrumGoerliNativeUSDC(token?.address)
+      ) {
+        setGasSummaryStatus('unavailable')
+        return
+      }
+    }
+
+    if (
+      arbTokenBridgeLoaded &&
+      arbTokenBridge &&
+      arbTokenBridge.eth &&
+      arbTokenBridge.token
+    ) {
+      estimateGas()
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    // Re-run gas estimation when:
+    estimateGas,
+    isDepositMode, // when user switches deposit/withdraw mode
+    amount, // when user changes the amount (check against the debounced value)
+    amountDebounced,
+    token, // when the token changes
+    l1Network.id, // when L1 and L2 network id changes
+    l2Network.id,
+    arbTokenBridgeLoaded,
+    walletAddress, // when user switches account or if user is not connected
+    setGasSummaryStatus
+  ])
+
+  useEffect(() => {
+    // rather than a deep equal, we are checking all the properties to see if they are sync
+
+    if (
+      gasSummary.estimatedL1GasFees !== estimatedL1GasFees ||
+      gasSummary.estimatedL2GasFees !== estimatedL2GasFees
+    ) {
+      setGasSummary({
+        estimatedL1GasFees,
+        estimatedL2GasFees
+      })
+    }
+  }, [
+    result,
+    isDepositMode,
+    l1GasPrice,
+    l2GasPrice,
+    gasSummary,
+    estimatedL1GasFees,
+    estimatedL2GasFees,
+    setGasSummary
+  ])
+}
+
+type GasSummaryStore = {
+  gasSummaryStatus: GasEstimationStatus
+  gasSummary: UseGasSummaryResult
+  setGasSummaryStatus: (gasSummaryStatus: GasEstimationStatus) => void
+  setGasSummary: (gasSummary: UseGasSummaryResult) => void
+}
+
+export const useGasSummaryStore = create<GasSummaryStore>()(set => ({
```

Comment:
Why do we need a store here? Do we plan to use this hook in multiple places? Otherwise, it would be fine to have the state inside the hook

Reply:
I had the very same opinion before looking at this comment.
Using `useGasSummaryStore` is just adding a layer of complexity. Since all the values in `useGasSummary()` are memoized, we should try and use that only.

The hook usage will also be simplified, as we won't need to call `useGasSummary()` and then get the computed values from `useGasSummaryResult()`.

Reply:
done, killed the `useGasSummaryStore`